src/compile/data/assemble.ts

```diff
@@ -200,6 +200,12 @@ function makeWalkTree(data: VgData[]) {
       dataSource.transform = dataSource.transform.concat(node.assemble());
     }
 
+    if (node instanceof AggregateNode) {
```

Comment:
Can you explain why we need this chunk of changes?

Reply:
Hmm, I just came across this code and still don't really get it. Why is this the right fix?

Reply:
Still don't remember why I added this chunk 1+ years ago but ran tests with and without the chunk and all pass.

Reply:
Thank you so much for looking into this. If we don't need this, let's remove it. Can you send a PR?

Reply:
#4178